### PR TITLE
[OBSDEF-12070] modified charts for decks-support-store

### PR DESCRIPTION
**Pull request by vgupta-mickey**
## Purpose
[OBSDEF-12070](https://jira.cec.lab.emc.com:8443/browse/OBSDEF-12070)

The decks-support-store is changed from StatefulSets to the Deployment for all the platforms except the VMware platform.
The reason is that by default the persistent volumes are generally backed up by a local disk and when the disk goes bad, the decks-support-store pod doesn't re-schedule and stays in a Pending state forever. 
In the case of the StatefulSets, when the pod is rec-created and it re-attaches to the newly created PVC.

For VMware,  the storage is backed by VSAN so we will continue using the deployment.

## PR checklist
- [x ] make test passed
- [x ] make build passed
- [ ] helm install <chart> passed
- [ ] vSphere7 make package deployments passed
- [ ] helm upgrade <chart> passed
- [ ] helm test <release_name> passed
- [ ] Passed - https://asd-ecs-jenkins.isus.emc.com/jenkins/view/ECS-Flex/job/charts-custom-ci/

## Testing
```
vinod@ubuntu:~/go/src/notes$ ssh root@10.244.130.29
root@10.244.130.29's password: 
Last login: Thu Sep 23 12:07:14 2021 from w1047ltk13.corp.emc.com
[root@centos77 ~]# lsblk
NAME            MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
fd0               2:0    1    4K  0 disk 
sda               8:0    0   30G  0 disk 
├─sda1            8:1    0    1G  0 part /boot
└─sda2            8:2    0   29G  0 part 
  ├─centos-root 253:0    0   26G  0 lvm  /
  └─centos-swap 253:1    0    3G  0 lvm  [SWAP]
sr0              11:0    1 1024M  0 rom  
[root@centos77 ~]# k get pods
NAME                                            READY   STATUS    RESTARTS   AGE
csi-baremetal-controller-5d57cf9f7-jrmsm        4/4     Running   0          81m
csi-baremetal-node-clfjw                        4/4     Running   0          81m
csi-baremetal-node-controller-cd86ffdbf-jr8st   1/1     Running   0          81m
csi-baremetal-node-g7lw7                        4/4     Running   0          81m
csi-baremetal-node-gfdtq                        4/4     Running   0          81m
csi-baremetal-node-nxrkb                        4/4     Running   0          81m
csi-baremetal-node-r2kgk                        4/4     Running   0          81m
csi-baremetal-node-rslqx                        4/4     Running   0          81m
csi-baremetal-node-t7fd2                        4/4     Running   0          81m
csi-baremetal-operator-5778bbf758-6fxw5         1/1     Running   0          83m
csi-baremetal-se-b5vkv                          1/1     Running   0          81m
csi-baremetal-se-g2nvb                          1/1     Running   0          81m
csi-baremetal-se-lpqzp                          1/1     Running   0          81m
[root@centos77 ~]# k get pods -n objectscale-sustem
No resources found in objectscale-sustem namespace.
[root@centos77 ~]# k get pods -n objectscale-system
NAME                                                              READY   STATUS    RESTARTS   AGE
atlas-operator-77f5b897d5-6zqcb                                   2/2     Running   0          73m
decks-6954865b6-xlp7w                                             1/1     Running   0          67m
decks-support-store-0                                             1/1     Running   0          67m
kahm-9f869ddf8-c6tdp                                              1/1     Running   0          68m
kahm-postgresql-ha-pgpool-d96fb475b-gjmqg                         1/1     Running   0          68m
kahm-postgresql-ha-pgpool-d96fb475b-j86fh                         1/1     Running   0          68m
kahm-postgresql-ha-postgresql-0                                   1/1     Running   0          68m
kahm-postgresql-ha-postgresql-1                                   1/1     Running   0          68m
kahm-postgresql-ha-postgresql-2                                   1/1     Running   0          68m
logging-injector-7b6fd874c6-85v24                                 2/2     Running   0          73m
objectscale-dcm-899456d9-xf62b                                    2/2     Running   0          73m
objectscale-dcm-atlas-0                                           2/2     Running   0          73m
objectscale-dcm-atlas-1                                           2/2     Running   0          73m
objectscale-dcm-atlas-2                                           2/2     Running   0          73m
objectscale-federation-85474877f-pwjtn                            2/2     Running   0          73m
objectscale-federation-85474877f-rg9dj                            2/2     Running   0          73m
objectscale-federation-85474877f-zcdr6                            2/2     Running   0          73m
objectscale-federation-atlas-0                                    2/2     Running   0          73m
objectscale-federation-atlas-1                                    2/2     Running   0          73m
objectscale-federation-atlas-2                                    2/2     Running   0          73m
objectscale-gateway-86cb865569-v6z2m                              1/1     Running   3          73m
objectscale-graphql-598468bd96-gjj5x                              2/2     Running   0          73m
objectscale-graphql-598468bd96-splvp                              2/2     Running   0          73m
objectscale-graphql-598468bd96-wrvtv                              2/2     Running   0          73m
objectscale-iam-7794cccc4f-b7w9j                                  2/2     Running   0          73m
objectscale-iam-7794cccc4f-m6v2z                                  2/2     Running   0          73m
objectscale-iam-7794cccc4f-v6wph                                  2/2     Running   0          73m
objectscale-iam-atlas-0                                           2/2     Running   0          73m
objectscale-iam-atlas-1                                           2/2     Running   0          73m
objectscale-iam-atlas-2                                           2/2     Running   0          73m
objectscale-install-controller-5c98b996cc-7xwjq                   1/1     Running   0          73m
objectscale-manager-alerts-6969465bdd-jqlw9                       4/4     Running   0          73m
objectscale-manager-bookkeeper-operator-7dbbd54cdc-vrhw6          2/2     Running   0          73m
objectscale-manager-fluxd-5c64bb78d6-tjkvj                        3/3     Running   0          73m
objectscale-manager-grafana-85f4dd9cfb-nz74d                      5/5     Running   0          73m
objectscale-manager-influxdb-0                                    5/5     Running   0          73m
objectscale-manager-influxdb-1                                    5/5     Running   0          73m
objectscale-manager-influxdb-2                                    5/5     Running   0          73m
objectscale-manager-influxdb-operator-8449d9c97-pp8hc             2/2     Running   0          73m
objectscale-manager-performance-1632404400-n8bvr                  1/1     Running   0          78s
objectscale-manager-pravega-operator-946cc8c4c-25gv8              2/2     Running   0          73m
objectscale-manager-rsyslog-7g42r                                 3/3     Running   0          73m
objectscale-manager-rsyslog-8z557                                 3/3     Running   0          73m
objectscale-manager-rsyslog-djbcm                                 3/3     Running   0          73m
objectscale-manager-rsyslog-j8tdv                                 3/3     Running   0          73m
objectscale-manager-rsyslog-n77vq                                 3/3     Running   0          73m
objectscale-manager-rsyslog-vc4ts                                 3/3     Running   0          73m
objectscale-manager-rsyslog-wdbmw                                 3/3     Running   0          73m
objectscale-manager-service-pod-66cf96654f-fn9qp                  1/1     Running   0          73m
objectscale-manager-statefuldaemonset-operator-6c56b85db8-vgkj5   2/2     Running   0          73m
objectscale-manager-telegraf-5f54695688-2vx5n                     3/3     Running   0          73m
objectscale-manager-throttler-6879c4d67b-hd6gv                    2/2     Running   0          73m
objectscale-operator-8696b854f4-xr7ww                             3/3     Running   0          73m
objectscale-portal-7d9c4f9556-bnr69                               2/2     Running   0          73m
objectscale-portal-7d9c4f9556-g7jcs                               2/2     Running   0          73m
objectscale-portal-7d9c4f9556-hwsws                               2/2     Running   0          73m
zookeeper-operator-786449c948-jnw77                               2/2     Running   0          73m
[root@centos77 ~]# k get pods -n objectscale-system |grep decks
decks-6954865b6-xlp7w                                             1/1     Running   0          67m
decks-support-store-0                                             1/1     Running   0          67m
[root@centos77 ~]# k describe pod decks-support-store-0 -n objectscale-system
Name:         decks-support-store-0
Namespace:    objectscale-system
Priority:     0
Node:         worker0.ocp4.dumpster.com/10.236.227.86
Start Time:   Thu, 23 Sep 2021 12:35:17 +0000
Labels:       controller-revision-hash=decks-support-store-65844564d9
              io.kompose.service=decks-support-store
              objectscale.dellemc.com/sp-integrated=true
              statefulset.kubernetes.io/pod-name=decks-support-store-0
Annotations:  appplatform.vmware.com/vsphere-emm-integrated: true
              k8s.v1.cni.cncf.io/network-status:
                [{
                    "name": "",
                    "interface": "eth0",
                    "ips": [
                        "10.254.5.70"
                    ],
                    "default": true,
                    "dns": {}
                }]
              k8s.v1.cni.cncf.io/networks-status:
                [{
                    "name": "",
                    "interface": "eth0",
                    "ips": [
                        "10.254.5.70"
                    ],
                    "default": true,
                    "dns": {}
                }]
              openshift.io/scc: anyuid
Status:       Running
IP:           10.254.5.70
IPs:
  IP:           10.254.5.70
Controlled By:  StatefulSet/decks-support-store
Containers:
  decks-support-store:
    Container ID:  cri-o://3dbf78135a9ff117f9ca606101aa1793dcc342d17bd9cb3f67cd046a946c3fde
    Image:         asdrepo.isus.emc.com:8099/decks-support-store:2.82.0-232.ff3325c
    Image ID:      asdrepo.isus.emc.com:8099/decks-support-store@sha256:7da60879415281c12095cb2c476fe444ec51f0c1df2a03aecfbffeaefe83f402
    Port:          7443/TCP
    Host Port:     0/TCP
    Command:
      /usr/local/bin/run-nginx.sh
    State:          Running
      Started:      Thu, 23 Sep 2021 12:35:22 +0000
    Ready:          True
    Restart Count:  0
    Environment:
      USERNAME:  <set to the key 'username' in secret 'decks-support-store-secret'>  Optional: false
      PASSWORD:  <set to the key 'password' in secret 'decks-support-store-secret'>  Optional: false
      SALT:      <set to the key 'salt' in secret 'decks-support-store-secret'>      Optional: false
    Mounts:
      /opt/secure/data from decks-support-store-claim0 (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-lgxvl (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             True 
  ContainersReady   True 
  PodScheduled      True 
Volumes:
  decks-support-store-claim0:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  decks-support-store-claim0-decks-support-store-0
    ReadOnly:   false
  default-token-lgxvl:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-lgxvl
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                 node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason            Age   From               Message
  ----     ------            ----  ----               -------
  Warning  FailedScheduling  67m   default-scheduler  0/7 nodes are available: .
  Warning  FailedScheduling  67m   default-scheduler  0/7 nodes are available: .
  Normal   Scheduled         66m   default-scheduler  Successfully assigned objectscale-system/decks-support-store-0 to worker0.ocp4.dumpster.com
  Normal   AddedInterface    66m   multus             Add eth0 [10.254.5.70/24]
  Normal   Pulling           66m   kubelet            Pulling image "asdrepo.isus.emc.com:8099/decks-support-store:2.82.0-232.ff3325c"
  Normal   Pulled            66m   kubelet            Successfully pulled image "asdrepo.isus.emc.com:8099/decks-support-store:2.82.0-232.ff3325c" in 1.31003711s
  Normal   Created           66m   kubelet            Created container decks-support-store
  Normal   Started           66m   kubelet            Started container decks-support-store
[root@centos77 ~]# k get pvc
No resources found in default namespace.
[root@centos77 ~]# k config set-context --current --namespace objectscale-system
Context "default/api-ocp4-dumpster-com:6443/system:admin" modified.
[root@centos77 ~]# k get pvc |grep decks
decks-support-store-claim0-decks-support-store-0                                         Bound    pvc-2f7a767b-27ba-45fb-b72a-0a1e4d9c167b   200Gi      RWO            csi-baremetal-sc-hddlvg   68m
[root@centos77 ~]# ssh worker0
Warning: the ECDSA host key for 'worker0.ocp4.dumpster.com' differs from the key for the IP address '10.236.227.86'
Offending key for IP in /root/.ssh/known_hosts:2
Matching host key in /root/.ssh/known_hosts:15

```

### CODE REVIEW

**Review by SarahAFochs — APPROVED** [review]

File: decks-support-store/templates/support-store-stateful-set.yaml [hunk 1]
```diff
@@ -0,0 +1,101 @@
+{{- if ne .Values.global.platform "VMware"}}
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: decks-support-store-headless
```
Comment (eddiepavkovic) [inline decks-support-store/templates/support-store-stateful-set.yaml:6]:
```suggestion
  name: decks-support-store
```
do we need `headless`  we don't have that for the deployment.

File: decks-support-store/templates/support-store-stateful-set.yaml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+metadata:
+  name: decks-support-store
+  namespace: {{.Release.Namespace}}
+  labels:
+    io.kompose.service: decks-support-store
+    app.kubernetes.io/name: decks
+    release: {{.Release.Name}}
+    name: decks
+spec:
+  serviceName: decks-support-store-headless
```
Comment (eddiepavkovic) [inline decks-support-store/templates/support-store-stateful-set.yaml:31]:
```suggestion
  serviceName: decks-support-store
```

File: decks-support-store/templates/support-store-stateful-set.yaml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  {{- if or .Values.global.storageClassName .Values.persistentVolume.storageClassName}}
+        storageClassName: {{ default .Values.global.storageClassName .Values.persistentVolume.storageClassName}}
+  {{- end}}
+        dataSource: null
+        resources:
+          requests:
+        {{- if .Values.persistentVolume.size}}
+            storage: {{ .Values.persistentVolume.size }} 
+        {{ else }}
+            storage: 200Gi
```
Comment (eddiepavkovic) [inline decks-support-store/templates/support-store-stateful-set.yaml:99]: simplify it to this to remove the if statement:
```
storage: {{ default "200Gi" .Values.persistentVolume.size }}
```

File: decks-support-store/templates/support-store-stateful-set.yaml (hunk 1 above)
Comment (vgupta-mickey) [reply decks-support-store/templates/support-store-stateful-set.yaml:6]: no..it has to be headless 
we have another service without headless

File: decks-support-store/templates/support-store-stateful-set.yaml (hunk 2 above)
Comment (vgupta-mickey) [reply decks-support-store/templates/support-store-stateful-set.yaml:31]: same

**Review by tmarcu — APPROVED** [review]

**Review by eddiepavkovic — APPROVED** [review]
cool... thanks g2g

**Pull request merged**